### ISSUE TITLE
Obtain container hostname from specified Docker container label

### ISSUE THREAD

**Issue by 2opremio** [weaveworks/weave#1381]
TL;DR Add a new flag to weave proxy `--hostname-from-label` which specifies a label key from which to extract the container's hostname.

We already added `--hostname-match` and `--hostname-replacement` to the proxy, allowing to derive hostnames from transformations of container names (see https://github.com/weaveworks/weave/pull/1126).

For ECS this is not enough since the original specified container names are [mangled destructively](https://github.com/weaveworks/guides/blob/5a70632985c5f1c64dcb807e35cb33cab6a7745e/aws-ecs/README.md#known-issues--limitations), not allowing to retrieve the original name at spawning time.

However, I just found out that ECS adds a label, `com.amazonaws.ecs.container-name`, to every container with the original container name.

We can add a new flag to weave proxy, `--hostname-from-label` to specifiy a label key from which to extract the container's hostname.

**Comment by rade** [comment]
> We can add a new flag to weave proxy, `--hostname-from-label` to specifiy a label key from which to extract the container's hostname.

And this could work in *conjunction* with the existing options, right?

**Comment by 2opremio** [comment]
Yes, I don't see a reason why it wouldn't.

**Comment by rade** [comment]
ok. hurry up, let's try to get this in 1.1 :)

**Comment by 2opremio** [comment]
I'm on it!

**Comment by faddat** [comment]
Can you give me an example of how I can use this?

**Comment by faddat** [comment]
Nevermind.  I did not know what docker labels could take key+value.  For the next ignorant sucker:

From the docker docs:

label (label=<key> or label=<key>=<value>)
